sys/crypto/modes/ccm.c

```diff
@@ -108,13 +108,17 @@ int ccm_compute_adata_mac(cipher_t* cipher, uint8_t* auth_data,
         /* 16 octet block size + max. 10 len encoding  */
         uint8_t auth_data_encoded[26], len_encoding = 0;
 
-        if ( auth_data_len < (((uint32_t) 2) << 16)) {       /* length (0x0001 ... 0xFEFF)  */
+        /* If 0 < l(a) < (2^16 - 2^8), then the length field is encoded as two
+         * octets. (RFC3610 page 2)
+         */
+        if (auth_data_len < (((uint32_t) 1) << 16) - (1 << 8)) {
```

Comment:
Minor nitpick, Is there a reason to use the bit shift notation here. I think I prefer the `0xFF00` notation here to match the table on page 4 of the RFC.

Reply:
I just kept the same notation as before to not change the way it was done.
I could even just put `<= 0xFEFF` as it is in the RFC too and match the comment.

Reply:
> I just kept the same notation as before to not change the way it was done.

I see.

>  I could even just put `<= 0xFEFF` as it is in the RFC too and match the comment.

Also fine with me :)

Reply:
I put ` <= 0xFEFF` now but could change if you prefer `0xFF00`.